metacluster/include/metacluster/ListTenants.actor.h

```diff
@@ -52,6 +52,28 @@ Future<std::vector<std::pair<TenantName, int64_t>>> listTenantsTransaction(Trans
 	    future);
 }
 
+ACTOR template <class Transaction>
+Future<std::vector<std::pair<TenantName, int64_t>>> listTenantGroupTenantsTransaction(Transaction tr,
+                                                                                      TenantName begin,
+                                                                                      TenantName end,
+                                                                                      int limit,
+                                                                                      TenantGroupName tenantGroup) {
+	tr->setOption(FDBTransactionOptions::RAW_ACCESS);
+	state KeyBackedSet<Tuple>::RangeResultType result = wait(metadata::management::tenantMetadata().tenantGroupTenantIndex.getRange(
+	    tr, Tuple::makeTuple(tenantGroup), Tuple::makeTuple(keyAfter(tenantGroup)), limit));
+	std::vector<std::pair<TenantName, int64_t>> returnResult;
+	if (!result.results.size()) {
+		return returnResult;
+	}
+	for (auto const& tupleEntry : result.results) {
+		TenantName tName = tupleEntry.getString(1);
+		if (tName >= begin && tName < end) {
+			returnResult.push_back(std::make_pair(tName, tupleEntry.getInt(2)));
```

Comment:
We should filter as part of the initial read, because otherwise we will be unable to read the tenants out of an especially large group. This also will not have the correct behavior for limit. The same is true for the tenant implementation.

Reply:
What is the right way to filter it as part of the initial read? Setting the other parts of the tuple like this as part of `metadata::management::tenantMetadata().tenantGroupTenantIndex.getRange` ?
```
Tuple::makeTuple(tenantGroup, begin)
Tuple::makeTuple(tenantGroup, end)
```
or is there another way?

Reply:
Yeah, the tenant name is the second element of the tuple, so you would set that item accordingly in your bounds